src/app/modules/submission/email-submission/email-submission.util.ts

```diff
@@ -518,3 +519,31 @@ export const concatAttachmentsAndResponses = (
   response += attachments.reduce((acc, { content }) => acc + content, '')
   return response
 }
+
+// Class to encapsulate maskedData method
+// which returns autoReplyData with masked Corppass UID
+
+export class autoReplyDataObj {
+  data: EmailAutoReplyField[]
+  constructor(autoReplyData: EmailAutoReplyField[]) {
+    this.data = autoReplyData
+  }
+  get maskedData(): EmailAutoReplyField[] {
+    const maskField = (field: EmailAutoReplyField) => {
+      const maskedAnswerTemplate = field.answerTemplate.map((answer) => {
+        return answer.length >= 4 // defensive, in case UID length is less than 4
+          ? '*'.repeat(answer.length - 4) + answer.substr(-4)
+          : answer
+      })
+      return {
+        question: field.question,
+        answerTemplate: maskedAnswerTemplate,
+      }
+    }
+    return this.data.map((autoReplyField) => {
+      return autoReplyField.question === SPCPValidatedFields.CpUid
```

Comment:
I'm not sure this is safe enough. anyone who creates a question called "CorpPass Validated UID" would get incorrectly masked email confirmations. can we add additional checks on the auth type of the form and the index of the field?

Reply:
I added checks for auth type and index. tbh, I'm not fully convinced checking for index is the right way to go. It is a coincidence that SPCPValidatedFields.CpUid happens to be pushed into parsedResponses array last and we shouldn't be relying on this. I added a comment to remind future devs in case they modify that part of the code, but it is not hard to imagine how this might be easily broken. 

Also, since we are already checking for AuthType, I can't see a use case where the admin decides to have two fields with the exact same question "CorpPass Validated UID".

Reply:
as spoken as syncup, I removed the check for index, but retained the check for authtype